server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java

```diff
@@ -60,21 +57,22 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.not;
 
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, transportClientRatio = 0)
 public abstract class AbstractDisruptionTestCase extends ESIntegTestCase {
 
     static final TimeValue DISRUPTION_HEALING_OVERHEAD = TimeValue.timeValueSeconds(40); // we use 30s as timeout in many places.
 
-    private NodeConfigurationSource discoveryConfig;
+    private Settings currentSettings;
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.builder().put(discoveryConfig.nodeSettings(nodeOrdinal))
+        return Settings.builder().put(currentSettings)
```

Comment:
Can we also defer to `super.nodeSettings(nodeOrdinal)` so we don't also need to set `DISCOVERY_HOSTS_PROVIDER_SETTING` and `MAX_LOCAL_STORAGE_NODES_SETTING` here? (This also picks up a correct value for `DISCOVERY_ZEN_PING_UNICAST_HOSTS_SETTING`).